### PR TITLE
Track mentions autocomplete opened

### PR DESCRIPTION
**Pull request by pelumy**
## Issues covered
#1480 

## Description
This PR tracks mentions when the mentions button is clicked and the user types an "@"

## How to test
1. Build the app.
2. Click on the post button at the bottom of the screen.
3. Type an "@" or press the mentions button.
4. Please check that the following is logged: "Mentions Opened".

### CODE REVIEW

**Comment by mplorentz** [comment]
👀

File: Nos/Service/Analytics.swift [hunk 1]
```diff
@@ -297,4 +297,8 @@ class Analytics {
     func cancelledUploadSourceSelection() {
         track("Cancelled Upload Source Selection")
     }
+
+    func openedMentions() {
+        track("Mentions Opened")
```
Comment (mplorentz) [inline Nos/Service/Analytics.swift:302]: How about "Mentions Autocomplete Opened"? I think it's a little more clear.

Comment (pelumy) [reply Nos/Service/Analytics.swift:302]: I wanted to use something anyone could easily understand on the dashboard, but  "Mentions Autocomplete Opened" definitely looks clearer.

**Comment by pelumy** [comment]
> This works really well, but I think there is some good refactoring to do:
> 
> * if you move the call to `analytics.openedMentions()` into a `didSet` action on `showMentionsAutocomplete` then it will be more resilient in the future.
> * Can you remove the old "Mentions Search Started" analytics event? It fires after the user starts typing, which isn't very useful (because it would miss if the user taps a row without typing).

Why didn't I think of the first refactoring? 🤦‍♀️ 
These make sense. Thank you.

**Comment by pelumy** [comment]
> Can you remove the old "Mentions Search Started" analytics event? It fires after the user starts typing, which isn't very useful (because it would miss if the user taps a row without typing)

@mplorentz , I think it's still kinda useful to see if users actually search when using the mentions autocomplete, or they just mention authors in the thread alone.

**Comment by mplorentz** [comment]
👀

**Review by mplorentz — APPROVED** [review]
Looks great, thanks!

**Pull request merged**